package/MDAnalysis/coordinates/XDR.py

```diff
@@ -141,9 +157,18 @@ def _reopen(self):
 
     def _read_frame(self, i):
         """read frame i"""
-        self._xdr.seek(i)
         self._frame = i - 1
-        return self._read_next_timestep()
+        try:
+            self._xdr.seek(i)
+            timestep = self._read_next_timestep()
+        except RuntimeError:
+            warnings.warn('seek failed, recalculating offsets and retry')
+            offsets = self._xdr.calc_offsets()
+            self._xdr.set_offsets(offsets)
+            self._read_offsets(store=True)
+            self._xdr.seek(i)
+            timestep = self._read_next_timestep()
+        return timestep
```

Comment:
This is the only method where I'm using the offsets. In the xdr library only `seek` and `__len__` use the offsets. For the other functions I'm just reading from whereever I'm in the xtc/trr, which is always a valid frame start if the xtc/trr isn't corrupted (or someone used the `_bytes*` method). There a read should fail normally.